### DISCUSSION TITLE
Fake date generator

### DISCUSSION THREAD

**Discussion by alexpts** [vectordotdev/vrl#494] [Q&A] [answered]
I want generate logs with fake date between {from: "iso8601", to "iso8601"}.
Can I make fake date in VRL script?

**Comment by jszwedko** [comment]
Hmm, I don't _think_ that would be possible via VRL at the moment. I'm curious what use case you have for this. Could you describe it more?

**Reply by alexpts** [reply]
I used `demo_logs` source for generate demo logs and VRL script for transfor. My logs has similar timestimp field. I want make logs for any days/months/years with generate any date via VRL script.

I use `vector` for seeding elasticsearch logs. All millions logs save at one day now. It is not good.


May it possible via convert `now()` to integer (millisec) and sub some random integer from 0 to xxx.  
Result convert to `timestamp` back.

**Reply by jszwedko** [reply, marked as answer]
Ah, gotcha. You certainly could convert the timestamp to a unix timestamp and use arithmetic. This is a bit on the fringes of Vector's use-cases so I'm not sure how much of a priority it would be to better support it, natively. You could also consider writing a shell script to generate the logs and feeding them into Vector via the `stdin` source to send them into Elasticsearch.

**Reply by alexpts** [reply]
Yes, I think about custom small generator logs before vector. Vector is great :). Source `demo_logs` demo_logs is good in general and I don’t want to write my own generator before vector.

Ok, I got the answer to my question. Thank you!
